app/scripts/controllers/reviewRegistration.js

```diff
@@ -419,10 +429,19 @@ angular
             question: 'Are you sure you want to delete this promotion?',
           })
           .then(function () {
-            var regCopy = angular.copy(currentRegistration);
-            _.remove(regCopy.promotions, { id: promoId });
+            const registrationCopy = angular.copy(currentRegistration);
+            const promotion = $scope.allPromotions.find(function (promotion) {
+              return promotion.id === promoId;
+            });
+
+            if (promotion.isGlobal) {
+              _.remove(registrationCopy.globalPromotions, { id: promoId });
+            } else {
+              _.remove(registrationCopy.promotions, { id: promoId });
+            }
+
             $http
-              .put('registrations/' + currentRegistration.id, regCopy)
+              .put('registrations/' + currentRegistration.id, registrationCopy)
```

Comment:
Does the server accept this? Does it ignore the extra fields we are adding to the registration?